### PR TITLE
Fix upload of nested file lists

### PR DESCRIPTION
**Pull request by DoctorJohn**
## Description

This PR is a followup to #979. Unfortunately I didn't consider nesting of file lists. This PR enables file lists to be nested in complex input structures and adds according tests to all integrations.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* None

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

This release fixes the upload of nested file lists. Among other use cases, having an
input type like shown below is now working properly.

```python
import typing
import strawberry
from strawberry.file_uploads import Upload


@strawberry.input
class FolderInput:
    files: typing.List[Upload]
```

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+```python
+import typing
+import strawberry
+from strawberry.file_uploads import Upload
+
+
+@strawberry.input
+class FolderInput:
+    files: typing.List[Upload]
```
Comment (patrick91) [inline RELEASE.md:14]: good catch! I haven't used file uploads much, so I didn't consider this use case 😊

**Review by patrick91 — APPROVED** [review]
Looks good to me!

File: RELEASE.md (hunk 1 above)
Comment (DoctorJohn) [reply RELEASE.md:14]: I also came across it only by chance while extending my python graphql clients tests to run against a Strawberry app :)

**Pull request merged**